### PR TITLE
support TLS in cdc components

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Support TLS in TiCDC, fulfill https://github.com/pingcap/ticdc/issues/250

### What is changed and how it works?

- Code changed:

    - upgrade PD dependency to `github.com/pingcap/pd/v4`
    - Add security config to `cdc server` and `cdc cli`
    - gRPC communication with PD/TiKV will be encrypted
    - HTTP status/metric server will be encrypted if security config is enabled
    - Redirect gRPC log to CDC 

- Test changed:
    
    - Add an integration test case with TLS enabled.

TODO:
- Encrypt Sink communication with MySQL/Kafka
- Support common name check

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Support TLS

### CODE REVIEW

File: tests/_utils/run_curl [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+TLS_DIR=$1
+shift 1
+
+if [ -z "${2-}" ]; then
+    POST_ARGS=
+else
+    POST_ARGS="--post-data $2"
+fi
+
+# FIXME: use `wget` instead of `curl` because the latter rejects ECC certs on our CI.
```
Comment (shafreeck) [inline tests/_utils/run_curl]: Is this FIXME still required?

File: cdc/kv/client.go [hunk 2]
```diff
@@ -945,6 +944,7 @@ func (s *eventFeedSession) receiveFromStream(
 
 	for {
 		cevent, err := stream.Recv()
+		log.Debug("recv events", zap.Reflect("events", cevent))
```
Comment (overvenus) [inline cdc/kv/client.go:947]:
```suggestion
```

File: cdc/kv/store_op.go [hunk 3]
```diff
@@ -18,6 +18,8 @@ import (
 
 	"github.com/pingcap/errors"
 	"github.com/pingcap/ticdc/pkg/flags"
+	"github.com/pingcap/ticdc/pkg/security"
+	"github.com/pingcap/tidb/config"
```
[the remark this answers is not in the document]
Comment (leoppro) [reply cdc/kv/store_op.go:22]: @overvenus

**Comment by overvenus** [comment]
> how about making the config as a global variable or setting config into context?

Generally speaking, using global variables is not good especially when it relates to security issues. Setting credentials in context is doable but it makes code less explicitly and it is very likely to leak credential to packages we don't want.

**Pull request merged**